File: macos/Timesdump/Timesdump/Models/TimestampResult.swift
import Foundation

struct TimestampResult: Identifiable, Equatable {
    let id = UUID()
    let originalValue: String
    let timestamp: TimeInterval
    let date: Date
    let formattedTime: String
    let isMilliseconds: Bool
    
    init(originalValue: String, timestamp: TimeInterval, date: Date, format: String, isMilliseconds: Bool) {
        self.originalValue = originalValue
        self.timestamp = timestamp
        self.date = date
        self.isMilliseconds = isMilliseconds
        
        let formatter = DateFormatter()
        formatter.dateFormat = format
        formatter.locale = Locale.current
        formatter.timeZone = TimeZone.current
        self.formattedTime = formatter.string(from: date)
    }
    
    var unitDescription: String {
        isMilliseconds 
            ? NSLocalizedString("result.unit.milliseconds", comment: "milliseconds")
            : NSLocalizedString("result.unit.seconds", comment: "seconds")
    }
}

File: windows/Timesdump/Models/TimestampResult.cs
namespace Timesdump.Models;

public class TimestampResult
{
    public string OriginalValue { get; }
    public double Timestamp { get; }
    public DateTime Date { get; }
    public string FormattedTime { get; }
    public bool IsMilliseconds { get; }

    public TimestampResult(string originalValue, double timestamp, DateTime date, string format, bool isMilliseconds)
    {
        OriginalValue = originalValue;
        Timestamp = timestamp;
        Date = date;
        IsMilliseconds = isMilliseconds;
        FormattedTime = date.ToString(format);
    }

    public string UnitDescription => IsMilliseconds
        ? LocalizationHelper.GetString("Result_Unit_Milliseconds")
        : LocalizationHelper.GetString("Result_Unit_Seconds");
}
